modin/core/storage_formats/pandas/native_query_compiler.py

```diff
@@ -585,7 +586,6 @@ class NativeQueryCompiler(BaseQueryCompiler):
     _shape_hint: Optional[str]
 
     def __init__(self, pandas_frame, shape_hint: Optional[str] = None):
-        assert NativeDataframeMode.get() == "Pandas"
```

Comment:
Why removed?

Reply:
Now constructor of NativeQueryCompiler can be also called when NativeDataframeMode.get() is default as even if the mode is set when we call this constructor in operations where the left operand is a NativeQueryCompiler we construct a new query compiler of type NativeQueryCompiler